### PR TITLE
Fix lipid spectrum chart

### PR DESCRIPTION
**Pull request by Annexhc**

### CODE REVIEW

File: src/main/java/io/github/mzmine/datamodel/features/types/annotations/LipidMatchListType.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
           createEntry(CommentType.class,
               match -> match.getComment() != null ? match.getComment() : ""),
           createEntry(LipidAnnotationMsMsScoreType.class, l -> l.getMsMsScore().floatValue()),
-          createEntry(LipidSpectrumType.class, match -> null), // ???
+          createEntry(LipidSpectrumType.class, match -> true), // ???
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/datamodel/features/types/annotations/LipidMatchListType.java:55]: Yes this was the most important change. Great that you spotted it.

File: src/main/java/io/github/mzmine/datamodel/features/types/graphicalnodes/LipidSpectrumChart.java [hunk 2]
```diff
@@ -23,119 +23,69 @@
...
-      // Add label generator for the dataset
-      MatchedLipidLabelGenerator matchedLipidLabelGenerator =
-          new MatchedLipidLabelGenerator(matchedFragments);
-      chart.getXYPlot().getRenderer().setSeriesItemLabelGenerator(
-          chart.getXYPlot().getSeriesCount(), matchedLipidLabelGenerator);
-
-      Platform.runLater(() -> chart.addDatasets(datasets));
-      setPrefHeight(GraphicalColumType.DEFAULT_GRAPHICAL_CELL_HEIGHT);
-      getChildren().add(chart);
+      Platform.runLater(() -> {
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/datamodel/features/types/graphicalnodes/LipidSpectrumChart.java:83]: We try to use the MZmineCore.runLater options. There is even one were you can force the thread to wait for it to be set.... but this is only important if your downstream stuff relies on it (like when changing the name of feature lists. Here its okay to use MZmineCore.runLater.

File: src/main/java/io/github/mzmine/gui/chartbasics/simplechart/providers/impl/spectra/LipidSpectrumProvider.java [hunk 3]
```diff
@@ -215,4 +184,14 @@ public int getValueCount() {
   public double getComputationFinishedPercentage() {
     return 1d;
   }
+
+  @Override
+  public @org.jetbrains.annotations.Nullable String getLabel(int index) {
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/gui/chartbasics/simplechart/providers/impl/spectra/LipidSpectrumProvider.java:189]: Actually replace all @nonnull with @NotNull and @Nullable from javax to jetbrains annotations:

Remove imports:
import javax.annotation.Nonnull;
import javax.annotation.Nullable;

And import the new annotations

**Comment by robinschmid** [comment]
Only a few items... @Annexhc can you apply those changes and I will merge it

**Comment by Annexhc** [comment]
@robinschmid all done, thanks for the review!

**Comment by robinschmid** [comment]
Thanks Harry

**Pull request merged**